internal/kafka/internal/services/kafka.go

```diff
@@ -427,6 +427,22 @@ func (k *kafkaService) RegisterKafkaJob(kafkaRequest *dbapi.KafkaRequest) *error
 
 	kafkaRequest.KafkaStorageSize = size.MaxDataRetentionSize.String()
 
+	// We intentionally manually set CreatedAt and UpdatedAt instead of letting
+	// gorm do it. The reason for that is that otherwise the ExpiresAt value
+	// would be calculated from a start date potentially earlier than the CreatedAt
+	// time a different value than those. An alternative would be performing
+	// two different database updates but it would be less performant
+	timeNow := dbConn.NowFunc()
+	// TODO should we manually set CreatedAt and UpdatedAt or just
+	// carry with the implication that ExpiresAt would be not calculated
+	// from the exact creation time but from a time potentially earlier than the
+	// creation time?
+	kafkaRequest.CreatedAt = timeNow
+	kafkaRequest.UpdatedAt = timeNow
+	if size.LifespanSeconds != nil {
+		kafkaRequest.ExpiresAt = sql.NullTime{Time: timeNow.Add(time.Duration(*size.LifespanSeconds) * time.Second), Valid: true}
+	}
```

Comment:
Whether we do it ourselves or let gorm do it, the difference will be milliseconds (or even less): it would be no harm.

I don't see any harm in this approach too. I've no preference for the two methods.

Reply:
:+1: I will leave it as it is now. This is, manually setting the createdat and updated at so they are consistent